src/lua/utils.c

```diff
@@ -499,6 +499,18 @@ luaT_call(struct lua_State *L, int nargs, int nreturns)
 	return 0;
 }
 
+int
+luaT_dostring(struct lua_State *L, const char *str)
+{
+	if (luaL_loadstring(L, str) != 0) {
+		diag_set(LuajitError, lua_tostring(L, -1));
```

Comment:
This function pollutes the Lua stack on error. We should probably pop the string to be on a safe side.

Anyway, I don't understand why you need this function. AFAIU in the scope of #7447 you need to run a log configuration function so the Lua syntax is guaranteed to be correct. Typically, we don't check the result of `luaL_loadstring` in such cases, but even if we did, a panic or assertion should be enough. Please merge it into the PR that fixes #7447 if you really need it.